src/nvim/ex_cmds.c

```diff
@@ -3638,6 +3638,13 @@ static int do_sub(exarg_T *eap, proftime_T timeout, long cmdpreview_ns, handle_T
     sub_needs_free = cmdpreview && sub != source;
   }
 
+  bool cmdheight0 = p_ch < 1 && !ui_has(kUIMessages);
```

Comment:
is this here because of cmdpreview? would it make sense to move this to `cmdpreview_show` ?

Reply:
This should be for the substitute prompt.

Reply:
This does not depend on inccommand preview.
It is for substitute prompt.

Please see this.

https://github.com/neovim/neovim/issues/18958